pkg/apis/sources/v1alpha1/apiserver_types.go

```diff
@@ -116,6 +116,11 @@ type ApiServerResource struct {
 	// +optional
 	LabelSelector *metav1.LabelSelector `json:"labelSelector"`
 
+	// ControllerSelector restricts this source to objects with the selected controller kind.
+	// Only apiVersion and kind are used. Both are optional
+	// +optional
+	ControllerSelector *metav1.OwnerReference `json:"controllerSelector"`
```

Comment:
How about a schema.GroupVersionKind instead? Or something else that does not include all the extraneous fields that we will either silently ignore or need to validate are empty.

Reply:
`schema.GroupVersionKind` does not have the JSON tags and I'd like to keep `apiVersion` to be consistent. Either way, extraneous fields are silently ignored until we decide to use k8s 1.15 structural schema.